### ISSUE TITLE
Disable pidfile using Puma::DSL

### ISSUE THREAD

**Issue by dmke** [puma/puma#3578]
**Is your feature request related to a problem? Please describe.**

I need to run a Rails application without Puma writing a PID file.

Using `pidfile nil` in `config/puma.rb` leads to an error on startup:

```console
$ bundle exec rails s -b :: -p 3000
=> Booting Puma
=> Rails 7.1.5 application starting in development 
=> Run `bin/rails server --help` for more startup options
Puma starting in single mode...
* Puma version: 6.5.0 ("Sky's Version")
* Ruby version: ruby 3.2.5 (2024-07-26 revision 31d0f1a2e7) [x86_64-linux]
*  Min threads: 5
*  Max threads: 5
*  Environment: development
*          PID: 52503
* Listening on http://[::]:3000
Exiting
GEM_HOME/puma-6.5.0/lib/puma/launcher.rb:316:in `write': No such file or directory @ rb_sysopen -  (Errno::ENOENT)
        from GEM_HOME/puma-6.5.0/lib/puma/launcher.rb:316:in `write_pid'
```

Using `pidfile '/dev/null'` leads to an error on shutdown:

```console
$ bundle exec rails s -b :: -p 3000
...
^C- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2024-12-13 10:54:18 +0100 ===
- Goodbye!
Exiting
GEM_HOME/puma-6.5.0/lib/puma/launcher.rb:143:in `unlink': Permission denied @ apply2files - /dev/null (Errno::EACCES)
        from GEM_HOME/puma-6.5.0/lib/puma/launcher.rb:143:in `delete_pidfile'
```

**Describe the solution you'd like**

Allow `pidfile nil` (or maybe `pidfile false` to better convey intend) in `Puma::DSL#pidfile`

https://github.com/puma/puma/blob/4d7f4e77bde20df51bfad4d8a1f290273c6b1fe7/lib/puma/dsl.rb#L456-L463

i.e. something like

```ruby
def pidfile(path)
  @options[:pidfile] = path == false ? path : path.to_s
end
```

Both `Puma::Launcher#write_pid` and `#delete_pidfile` haven a guard clause to protect against the PID file path being falsy.

**Describe alternatives you've considered**

I'm currently using this workaround:

```ruby
# RAILS_ROOT/config/puma.rb

if ENV.fetch("RAILS_ENV", "development") == "production"
  pidfile Struct.new(:hack) {
    def to_s = nil
  }.new
end
```

**Additional context**

- Puma 6.5.0
- Rails 7.1.5
- Ruby 3.2.5

Since this is Rails-specific, I've dug out the following:

- rails/rails#52558
- [`Rails::Command::ServerCommand#pid`](https://github.com/rails/rails/blob/14c115b120ed089331ff3dc13f36bd9129ced33d/railties/lib/rails/commands/server/server_command.rb#L245-L247), which returns the CLI arg, or `ENV["PIDFILE"]`, or the default path (`tmp/pids/server.pid`).

They seem to be reluctant to change their design. The argument is that "PID files in development are useful for the LSP".

**Comment by dmke** [comment]
I'm realizing that my issue seems to be fixed with Rails v7.2+ (see https://github.com/rails/rails/commit/482330d156ebde147992c1973b4c572cee89737e)

I'm not sure how fast I can upgrade the application though (there are some other dependencies not yet ready for Rails 7.2...)

**Comment by dmke** [comment]
I'm closing this, Puma works as intended.

The PID file I'm seeing is set by Rails. When omitting `pidfile` in the config file, Puma does not create a PID file on its own. It doesn't make much sense to have an option for disabling a knob, when the solution is to simply not use that knob :)

Since I currently can't upgrade the Rails app to use v7.2+, I've changed the deployment so that it doesn't execute `rails server`, but `puma -C config/puma.rb`. This circumvents the defaults Rails imposes.
